internal/runners/hello/hello_example.go

```diff
@@ -58,29 +56,48 @@ func New(p primeable) *Hello {
 	}
 }
 
+// processError contains the scope in which errors are processed. This is
+// useful for ensuring that errors are wrapped in a user-facing error and
+// localized.
+func processError(err *error) {
+	if err != nil {
+		switch {
+		case errs.Matches(*err, &runbits.NoNameProvidedError{}):
+			// Errors that we are looking for should be wrapped in a user-facing error.
+			// Ensure we wrap the top-level error returned from the runner and not
+			// the unpacked error that we are inspecting.
+			*err = errs.WrapUserFacingError(*err, locale.Tl("hello_err_no_name", "Cannot say hello because no name was provided."))
+		case errs.Matches(*err, &errs.ErrNoProject{}):
+			werr := errs.WrapUserFacingError(*err, locale.Tl("hello_err_no_project", "Cannot say hello because you are not in a project directory."))
+
+			// It's useful to offer users reasonable tips on recourses.
+			*err = errs.AddTips(werr, locale.Tl(
+				"hello_suggest_checkout",
+				"Try using [ACTIONABLE]`state checkout`[/RESET] first.",
+			))
```

Comment:
This feels like it would be a common thing we'd want to do. How about we make the last arg for `WrapUserFacingError()` take tips?